network.go

```diff
@@ -185,7 +225,7 @@ func (s *sandbox) updateInterface(netHandle *netlink.Handle, iface *pb.Interface
 	defer s.network.ifacesLock.Unlock()
 
 	if iface == nil {
-		return nil, grpcStatus.Errorf(codes.InvalidArgument, "Provided interface is nil")
+		return nil, errNoIF
```

Comment:
Maybe this is also better to move outside above the lock? So we can avoid an unnecessary lock

Reply:
Good catch! Branch updated.